Source/utils/soundsample.h

```diff
@@ -13,11 +16,20 @@ class SoundSample final {
 	void Play(int lVolume, int lPan, int channel = -1);
 	void Stop();
 	int SetChunkStream(HANDLE stormHandle);
-	int SetChunk(BYTE *fileData, DWORD dwBytes);
+
+	/**
+	 * @brief Sets the sample's WAV, FLAC, or Ogg/Vorbis data.
+	 * @param fileData Buffer containing the data
+	 * @param dwBytes Length of buffer
+	 * @return 0 on success, -1 otherwise
+	 */
+	int SetChunk(std::unique_ptr<std::byte[]> file_data, DWORD dwBytes);
+
 	int GetLength();
 
 private:
-	Mix_Chunk *chunk;
+	std::unique_ptr<std::byte[]> file_data_;

```

Comment:
I'd use a std::vector<std::byte> for this. No particular reason other than its's easier to handle and understand than a unique_ptr of bytes, and you don't have to lug around the size separately.

Reply:
the size is known in advance and never changes, vector is wrong / overkill / miscommunicates intent for this